test/e2e/util.go

```diff
@@ -3753,3 +3767,29 @@ func CoreDump(dir string) {
 		Logf("Error running cluster/log-dump.sh: %v", err)
 	}
 }
+
+// Run a function over each pod in a namespace.
+// Meant to be called in a loop - where the fn does some assertion logic on a test (i.e. look for string in the pod logs)
+func TryEachPod(c *client.Client, ns, selectorKey, selectorValue string, fn func(api.Pod)) {
+	pods := []api.Pod{}
+	for t := time.Now(); time.Since(t) < podListTimeout; time.Sleep(poll) {
```

Comment:
This particular function selects pods which don't have an associated RC and re run's a lambda on them.  Other functions... Its a little unique... I guess you're asking "why not wait for all pods running?"...

- Because it allows for nice realtime feedback.  Waiting for all pods running doesn't really tell us much when running on a large cluster (30,40,...,nodes).
- We also want to run this function even against pods that aren't running (i.e. PodSucceeded state). Remember that a GB of so can easily be sent quickly on a modern network.  So many iperf pods will simply complete in a few seconds, we want to continue polling them and keep updating results, not just wait till the slowest one is finished.

- I agree with you that the number of "do this a bunch of times with these pods" functions is getting astronomical.  I think there are like 25 different ones or something, in util.go.    

We definetely need to audit them for redundancy's  in  #23540, in my case at first glance i don't see anything that does exactly what i need here, but maybe i can modularize one of them.